WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsListViewModel.swift

```diff
@@ -120,22 +155,38 @@ private extension CustomFieldsListViewModel {
         } else {
             addedFields.append(field)
         }
-
-        updateCombinedList()
     }
 
-    func updateCombinedList() {
-        combinedList = originalCustomFields
-            .filter { field in !deletedFieldIds.contains(where: { $0 == field.id }) }
-            .map { field in editedFields.first(where: { $0.fieldId == field.id }) ?? CustomFieldUI(customField: field) }
-            + addedFields
-    }
+    func observePendingChanges() {
+        $pendingChanges
+            .map { [weak self] pendingChanges in
+                guard let self = self else { return [] }
+
+                return self.originalCustomFields
+                    .filter { field in !pendingChanges.deletedFieldIds.contains(where: { $0 == field.id }) }
+                    .map { field in pendingChanges.editedFields.first(where: { $0.fieldId == field.id }) ?? CustomFieldUI(customField: field) }
+                    + pendingChanges.addedFields
+            }
+            .assign(to: &$combinedList)
```

Comment:
I added this to make the updates to the combinedList in a reactive way, this avoids having to think about calling `updateCombinedList` everywhere.

Reply:
Good point, done in 009c8c2